linera-execution/src/system.rs

```diff
@@ -652,10 +644,73 @@ where
+                self.blob_published(&BlobId::new(blob_hash, BlobType::Committee))?;
+            }
             ReadBlob { blob_id } => {
                 self.read_blob_content(blob_id).await?;
                 self.blob_used(Some(txn_tracker), blob_id).await?;
             }
+            ProcessNewEpoch(epoch) => {
+                let chain_next_epoch = self.epoch.get().expect("chain is active").try_add_one()?;
+                ensure!(
+                    epoch == chain_next_epoch,
+                    SystemExecutionError::InvalidCommitteeCreation
+                );
+                let admin_id = self
+                    .admin_id
+                    .get()
+                    .ok_or_else(|| SystemExecutionError::InactiveChain)?;
+                let event_id = EventId {
+                    chain_id: admin_id,
+                    stream_id: StreamId::system(EPOCH_STREAM_NAME),
+                    key: bcs::to_bytes(&epoch)?,
+                };
+                let bytes = match txn_tracker.next_replayed_oracle_response()? {
+                    None => self.context().extra().get_event(event_id.clone()).await?,
+                    Some(OracleResponse::Event(recorded_event_id, bytes))
+                        if recorded_event_id == event_id =>
+                    {
+                        bytes
+                    }
+                    Some(_) => return Err(SystemExecutionError::OracleResponseMismatch),
+                };
+                let blob_id = BlobId::new(bcs::from_bytes(&bytes)?, BlobType::Committee);
+                txn_tracker.add_oracle_response(OracleResponse::Event(event_id, bytes));
+                let committee = bcs::from_bytes(self.read_blob_content(blob_id).await?.bytes())?;
+                self.blob_used(Some(txn_tracker), blob_id).await?;
+                self.committees.get_mut().insert(epoch, committee);
+                self.epoch.set(Some(epoch));
+            }
+            ProcessRemovedEpoch(epoch) => {
+                let chain_epoch = self.epoch.get().expect("chain is active");
+                ensure!(
+                    epoch < chain_epoch,
+                    SystemExecutionError::InvalidCommitteeRemoval
+                );
+                let admin_id = self
+                    .admin_id
+                    .get()
+                    .ok_or_else(|| SystemExecutionError::InactiveChain)?;
+                let event_id = EventId {
+                    chain_id: admin_id,
+                    stream_id: StreamId::system(REMOVED_EPOCH_STREAM_NAME),
+                    key: bcs::to_bytes(&epoch)?,
+                };
+                let bytes = match txn_tracker.next_replayed_oracle_response()? {
+                    None => self.context().extra().get_event(event_id.clone()).await?,
+                    Some(OracleResponse::Event(recorded_event_id, bytes))
+                        if recorded_event_id == event_id =>
+                    {
```

Comment:
Same here.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3432/.